content/200-concepts/100-components/02-prisma-client/260-client-extensions/120-query.mdx

```diff
@@ -0,0 +1,183 @@
+
+- `model`: the name of the containing model for the query that we want to extend.
+
+  In the above example, the `model` is a string of type `"User"`.
+
+- `operation`: the name of the operation being extended and executed.
+
+  In the above example, the `operation` is a string of type `"findMany"`.
+
+- `args`: the specific query input information to be extended.
+
+  This is a type-safe object that you can mutate before the query happens. You can mutate any of the properties in `args`. Exception: you cannot mutate `include` or `select` because that would change the expected output type and break type safety.
+
+- `query`: a promise for the result of the query.
+
+  - You can use `await` and then mutate the result of this promise, because its value is type-safe. TypeScript catches any unsafe mutations on the object.
+
+### Add or modify a custom query for a specific operation in all models of your schema
+
+To extend the queries in all the models of your schema, use `$allModels` instead of a specific model name. For example:
+
+```ts
+const xprisma = prisma.$extends({
+  query: {
+    $allModels: {
+      async findMany({ model, operation, args, query }) {
+        // set `age` and fill with the rest of `where`
+        args.where = { age: { gt: 18 }, ...args.where }
+        const users = await query(args)
+        return query(args)
+      },
+    },
+  },
+})
+```
+
+### Add or modify a custom query for all operations in a specific model
+
+Use `$allOperations` to extend all operations in a specific model.
+
+For example, the following code applies a custom query to all operations on the `user` model:
+
+```ts
+const xprisma = prisma.$extends({
+  query: {
+    user: {
+      $allOperations({ model, operation, args, query }) {
+        // handle all operations
+        return query(args)
+      },
+    },
+  },
+})
+```
+
+### Add or modify a custom query for all operations in all models of your schema
+
+Use `$allOperations` to extend all operations in all models of your schema.
+
+To apply a custom query to all operations on all models of your schema:
+
+```ts
+const xprisma = prisma.$extends({
+  query: {
+    $allModels: {
+      $allOperations({ model, operation, args, query }) {
+        // handle all operations
+        return query(args)
+      },
+    },
+  },
+})
+```
+
+## Wrap a query into a batch transaction
+
+You can wrap your extended queries into a [batch transaction](/guides/performance-and-optimization/prisma-client-transactions-guide). An example use-case is to enact row-level security (RLS).
+
+```ts
+const xprisma = prisma.$extends({
+  query: {
+    user: {
+      async findFirst({ args, query, operation }) {
+        return (await prisma.$transaction([query(args)]))[0]
+      },
+    },
+  },
+})
+```
```

Comment:
Hm, this is weird on that headline level. Is that just one very specific example on how to use a query extension component?

Reply:
I think we need to provide better context for this example and/or replace or expand the example. I'll follow this up later in the week.